### DISCUSSION TITLE
Increase the likeliness of generating an unique "stats_id"

### DISCUSSION THREAD

**Discussion by jonasraoni** [pkp/pkp-lib#7988] [Accepted Proposals]
Given the [uniqid](https://www.php.net/manual/en/function.uniqid.php#95001) function isn't that great for the task, perhaps we could use a package to generate GUIDs (e.g. https://packagist.org/packages/ramsey/uuid).

**Comment by asmecher** [comment]
uniqid results are typically 23 characters long (2^184 combinations) and stats IDs are created once per OJS/OMP/OPS installation; is this actually a problem?

**Comment by jonasraoni** [comment]
I think the uniqid is ok, so it's just a safer suggestion, GUIDs are great for such cases (banking systems normally use them), have a fixed length, etc., I can't see drawbacks 🤔

Another cool use is to replace auto-incrementing IDs, which have its pros (we don't even need to request the "inserted ID" from the database, as the ID can be generated by the client, naturally obfuscated, ...) and cons (bigger size, not easy to be sorted, ...).

**Comment by jonasraoni** [comment]
The GUID has been implemented at https://github.com/pkp/pkp-lib/issues/9136, see: https://github.com/pkp/pkp-lib/pull/9232/files#diff-a90b36dbf19885494106c6ec34cc0204c08f1b195de2d93abd68d694e963d53bR290

**Comment by jonasraoni** [comment]
ps:  I don't have permission to change the category to "Accepted Proposal" 😁

**Comment by asmecher** [comment]
Changed!
